packages/samples/headless-react/src/components/facet-manager/facet-manager.class.tsx

```diff
@@ -8,7 +8,13 @@ import {
 import {Children, Component, ContextType, ReactElement} from 'react';
 import {AppContext} from '../../context/engine';
 
-type FacetManagerChild = ReactElement<{facetId: string}>;
+export interface FacetManagerClassChildProps {
+  controllerOptions: {facetId: string};
+  /** @deprecated */
+  facetId?: string;
```

Comment:
Would be removed when all components are converted.

Reply:
Yeah, for the sample, I'm not sure we should bother with any deprecation notice...

I think it's fair to just update them/change them completely in a minor version, unless there is something I'm not seeing.

Reply:
yeah, what I meant is that the final PR which would be merged wouldn't have that. The `@deprecated` here is basically a TODO for myself.